esp-hal-embassy/src/lib.rs

```diff
@@ -75,69 +73,87 @@ macro_rules! mk_static {
     }};
 }
 
-/// A trait to allow better UX for initializing the timers.
-///
-/// This trait is meant to be used only for the `init` function.
-/// Calling `timers()` multiple times may panic.
-pub trait TimerCollection {
-    /// Returns the timers as a slice.
-    fn timers(self) -> &'static mut [Timer];
+mod private {
```

Comment:
I wonder how many we have of these before we should split it out into a seperate crate like esp-hal-internal :thinking:. This is fine as is for now though.